test.js

```diff
@@ -1,51 +1,100 @@
+const requestedHost = '127.0.0.1';
+const defaultHost = '0.0.0.0';
+
+const wrapper = (t, run) => new Promise(resolve => run(t, resolve));
+
+test.serial('should return the passed port value', async (t) => {
   const port = await findFreePort(3000);
 
   t.is(port, 3000);
 });
 
-test('should throw a port searching error when finding port outside the available range', async (t) => {
+test.serial('should throw a port searching error when finding port outside the available range', async (t) => {
   const port = await t.throwsAsync(() => findFreePort(65536));
 
   t.true(port.message.includes('should be >= 0 and < 65536. Received 65536.'));
 });
 
-test('should return `true` when available port is chosen', async (t) => {
+test.serial('should return `true` when available port is chosen', async (t) => {
   const port = await isPortFree(3007);
 
   t.is(port, true);
 });
 
-test('should return `false` when occupied is chosen', async (t) => {
-  const needfulPort = 3010;
+test.serial('should throw a port searching error when checking port outside the available range', async (t) => {
+  const port = await t.throwsAsync(() => isPortFree(65536));
+
+  t.true(port.message.includes('should be >= 0 and < 65536. Received 65536.'));
+});
+
+test.serial('should return the passed port value of the requested host', async (t) => {
+  const port = await findFreePort(3000, '127.0.0.1');
 
+  t.is(port, 3000);
+});
+
+test.serial('should return `false` when occupied port is chosen', wrapper, (t, done) => {
   const server = net.createServer();
-  await pify(server.listen.bind(server))(needfulPort, '0.0.0.0');
 
-  const port = await isPortFree(needfulPort);
+  server.once('listening', async () => {
+    server.once('close', () => setTimeout(done, 3000));
 
-  t.is(port, false);
+    const port = await isPortFree(requestedPort);
+    t.is(port, false);
+
+    server.close();
+  });
+
+  server.listen(requestedPort, defaultHost);
 });
 
-test('should throw a port searching error when checking port outside the available range', async (t) => {
-  const port = await t.throwsAsync(() => isPortFree(65536));
+test.serial('should return first available port', wrapper, (t, done) => {
+  const server = net.createServer();
 
-  t.true(port.message.includes('should be >= 0 and < 65536. Received 65536.'));
+  server.once('listening', async () => {
+    server.once('close', () => setTimeout(done, 3000));
+
+    const port = await findFreePort(requestedPort);
+    t.not(port, requestedPort);
+
+    server.close();
+  });
+
+  server.listen(requestedPort, defaultHost);
 });
 
-test('should return first available port', async (t) => {
-  const needfulPort = 3005;
+test.serial('should return `false` when occupied port of a requested host is chosen', wrapper, (t, done) => {
+  const server = net.createServer();
+
+  server.once('listening', async () => {
+    server.once('close', () => setTimeout(done, 3000));
 
+    const port = await isPortFree(requestedPort, requestedHost);
```

Comment:
Well, actually it's not a port, it's a `isFree` flag.

Reply:
Renamed constants.

Reply:
👍